File: app/src/main/java/com/example/pam_app/di/ContainerLocator.java
package com.example.pam_app.di;

import android.content.Context;

public class ContainerLocator {

    private static Container container;

    private ContainerLocator() {
        // empty
    }

    public static Container locateComponent(final Context context) {
        if (container == null) {
            setComponent(new ProductionContainer(context));
        }

        return container;
    }

    public static void setComponent(final Container container) {
        ContainerLocator.container = container;
    }
}

File: app/src/main/java/com/example/pam_app/di/ContainerLocator.kt
package com.example.pam_app.di

import android.content.Context
import kotlin.jvm.Synchronized

object ContainerLocator {
    private var container: Container? = null
    fun locateComponent(context: Context?): Container? {
        if (container == null) {
            setComponent(ProductionContainer(context))
        }
        return container
    }

    fun setComponent(container: Container?) {
        ContainerLocator.container = container
    }
}